tests/conversion

```
@@ -0,0 +1,264 @@
+    else
+        echo "===> FAIL: IPv6 address: ${instName}"
+        fail=1
+    fi
+
+    return "${fail}"
+}
+
+checkDNS() {
+    instName=$1
+
+    # Test lxd-agent and DNS resolution.
+    if lxc exec "${instName}" -- nslookup canonical.com >/dev/null 2>&1; then
+        echo "===> PASS: DNS resolution: ${instName}"
+        return 0
+    fi
+
+    if lxc exec "${instName}" -- getent hosts canonical.com >/dev/null 2>&1; then
+        echo "===> PASS: DNS resolution: ${instName}"
+        return 0
+    fi
+
+    echo "===> FAIL: DNS resolution: ${instName}"
+    return 1
+}
+
+# conversion_vm creates a new LXD virtual machine from the image on the given path.
+# A dedicated storage pool of the given type will be created for a new virtual machine.
+#
+# Input arguments:
+# 1: Name of new virtual machine.
+# 2: Type of the storage pool.
+# 3: Path to the image.
+conversion_vm() {
+    conversion vm "$1" "$2" "$3"
+}
+
+# conversion_container creates a new LXD container from the filesystem of the existing container.
+#
+# Input arguments:
+# 1: Name of the existing container.
+conversion_container() {
+    conversion container "$1" "" "/"
+}
+
+# conversion runs lxd-migrate for the given image path. For virtual machine, lxd-migrate is
+# executed on the localhost. For container, lxd-migrate will be installed and executed from
+# an existing container.
+conversion() {
+    instType=$1
+    instName=$2
+    poolType=$3
+    imgPath=$4
+
+    if [ "${instType}" = "vm" ]; then
+        echo "==> TEST: Conversion: Import virtual-machine '${instName}' on '${poolType}' storage pool"
+
+        poolName="conversion-${poolType}"
+        hostAddr="127.0.0.1"
+        instTypeCode="2" # VM in migration questioneer.
+
+        lxdMigrateCmd="lxd-migrate --conversion format"
+
+        # Ensure instance and storage pool with the given name do not exist.
+        lxc delete -f "${instName}" 2>/dev/null || true
+        lxc storage delete "${poolName}" 2>/dev/null || true
```

Why suppress stderr and ignore errors? Is this actually needed considering the cleanup at the end of the function?

---

I used that for cleanup during development to ensure those resources are wiped before running the script. Have removed this no longer necessary cleanup.